### DISCUSSION TITLE
Powers of ten and unit prefixes

### DISCUSSION THREAD

**Discussion by RudolfSch** [PaddiM8/kalk#136] [Ideas]
1. In engineering numbers are typically entered with powers of ten. Other calculators use a notation like this
1.23E3   for 1230 
1.23E-3 for 0.00123 etc.
How can we enter this form in kalker? I didn't find a hint in the documentation or have I overlooked something? 
Certainly 1.23*10^-3 would be possible, but it's a bit laborious.

2. Is there something like scientific and engineering mode in kalker? 
https://math.stackexchange.com/questions/1472020/calculator-mode-eng-mode-sci-vs-flo-vs-eng

3. Would it be somehow possible to enable (the most important = T to p) unit prefixes in kalker?
https://en.wikipedia.org/wiki/Unit_prefix
With this we could enter 1.2k for 1200 or 1.2u for 0.0000012 

These changes or some cool workarounds would be extremely helpful in a day to day life of an engineer using kalker. Do you see any chance of this being implemented?

**Comment by PaddiM8** [comment]
I've been asked about E-notation before and I've been unsure about how do deal with it, since 1.23E3 would currently be parsed as `1.23*E*3`, and people may expect to be able to define a variable called `E`. It would be a bit confusing, but I guess it might be fine? If `1.23E` is parsed as `1.23*E` and `1.23E3` is parsed as E-notation, it would probably be flexible enough. I think I'll add this!

Hadn't heard of engineering mode before but it sounds reasonable so I'll add that to my todo list.

Those kind of unit prefixes would be possible right now, but you have to add them yourself, since having too many pre-defined things would be confusing for people.

```
k = 1000
u = 10^-6

1.2k
1.2u
```

Thanks for the suggestions!

**Reply by RudolfSch** [reply]
E-notation - oh that would be great and certainly highly appreciated by many.

For reference: I added these lines to the default.kalker file
P = 10^15
T = 10^12
G = 10^9
M = 10^6
k = 10^3
m = 10^-3
u = 10^-6
n = 10^-9
p = 10^-12

**Reply by PaddiM8** [reply]
I implemented some of the things we talked about now https://github.com/PaddiM8/kalker/releases/tag/v2.1.0

Hopefully they work as expected!
